### DISCUSSION TITLE
Integration with Electron

### DISCUSSION THREAD

**Discussion by abelta** [mswjs/msw#2384] [Q&A]
Does MSW work at all with Electron? Anybody has a recent tutorial?
I'm starting a **Forge + Electron + React** project and I'm going through hoops and loops trying to figure out why it doesn't work because the usual type of configuration doesn't work without heavy tweaking.

* So far I've had to use CopyWebpackPlugin to copy mockServiceWorker into the .webpack/rendered folder
* tweak the html tag with the Content-Security-Policy (CPS)
* explicitly configure devContentSecurityPolicy in the forge-webpack configuration because none tell you this but it is rewritten by webpack

All of these changes I've found scattered all through the Internet but there doesn't seem to be a full fledged tutorial and I feel like I'm closer every time but so far I haven't been able to mock a single call. They are either caught by the CPS or they fail in the service worker.

I wouldn't be surprised if MSW was outright incompatible with Electron but even that would be nice to know.

**Comment by kettanaito** [comment]
Hi, @abelta. 

As I recall, you need to set `nodeIntegration: true` in your Electron config to enable MSW. Make sure to only set it in development as they settings seems to have security repercussions, and you shouldn't be using it in production, much like MSW itself. 

No other special configuration should be necessary. 

Then, you integrate MSW in your React application as you normally do, following the [Browser integration](https://mswjs.io/docs/integrations/browser) guidelines.
